packages/access-client/src/agent.js

```diff
@@ -254,17 +255,17 @@ export class Agent {
    * proofs matching the passed capabilities require it.
    *
    * @param {import('@ucanto/interface').Capability[]} [caps] - Capabilities to filter by. Empty or undefined caps with return all the proofs.
+   * @param {Ucanto.DID} [invocationAudience] - audience of invocation these proofs will be bundled with.
    */
-  proofs(caps) {
+  proofs(caps, invocationAudience) {
```

Comment:
this is a bit confusing to me - we might be getting proofs for a delegation that we want to issue to someone else, in which case there won't be an invocation at all - I'd at least expect this to be an optional parameter, but at a higher level it just feels confusing to be talking about invocations here at all...

Reply:
We spoke after this comment. I mentioned I had been trying to write this as an optional parameter. I've since made commits to make it more obvious that undefined is allowed, and based on @travis suggestion I also changed this second positional param to be an options object with options.sessionProofIssuer filter that can be used when `Agent.invoke` calls `Agent.proofs`

Reply:
Later I ended up just removing this new param alltogether. Now there is no API change that allows for filtering the proofs from the `proofs` method to only get proofs related to building a certain invocation. The tradeoff is that `proofs()` may return more than is strictly needed, but @travis explained to me in 1:1 that that is a good way to err for now (and as long as we have tests for tricky scenarios, we can always refactor to optimize invocation/proofs size later without regressions)